### DISCUSSION TITLE
Performance problem on form.trigger with nested fields

### DISCUSSION THREAD

**Discussion by joerex** [react-hook-form/react-hook-form#7998] [Q&A] [answered]
I was seeing some bad lag when executing `form.tigger()` on my form (of minimal complexity) and the way I was able to resolve the problem was by removing some unessesacry nesting. I had used a namespace for some fields so they would be easier to debug like so: `files.<file-id>.format`. When I use the trigger function the window would freeze for a few seconds. Removing the `files` prefix solved the problem. Is there an easy explanation for this? From the docs it seems like there shouldn't be a problem with nested fields.

Thanks for the awesome library @bluebill1049 !

**Comment by bluebill1049** [comment, marked as answer]
couple things:

- trigger() will run the entire form validation 
- trigger('singleField') is always the fastest or combine with forEach
- the deeper the nested field the worse performance is going to be bothered for type checking and run time, due to more iteration.
